synapseclient/models/protocols/file_protocol.py

```diff
@@ -0,0 +1,274 @@
+
+    def store(
+        self,
+        parent: Optional[Union["Folder", "Project"]] = None,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "File":
+        """
+        Store the file in Synapse. With this method you may:
+
+        - Upload a file into Synapse
+        - Update the metadata of a file in Synapse
+        - Store a File object in Synapse without updating a file by setting
+            `synapse_store` to False.
+        - Change the name of a file in Synapse by setting the `name` attribute of the
+            File object. Also see the [synapseclient.models.File.change_metadata][]
+            method for changing the name of the downloaded file.
+        - Moving a file to a new parent by setting the `parent_id` attribute of the
+            File object.
+
+        Arguments:
+            parent: The parent folder or project to store the file in. May also be
+                specified in the File object. If both are provided the parent passed
+                into `store` will take precedence.
+            synapse_client: If not passed in or None this will use the last client from
+                the `.login()` method.
+
+        Returns:
+            The file object.
+
+
+        Example: Using this function
+            File with the ID `syn123` at path `path/to/file.txt`:
+
+                file_instance = File(id="syn123", path="path/to/file.txt").store()
+
+            File at the path `path/to/file.txt` and a parent folder with the ID `syn456`:
+
+                file_instance = File(path="path/to/file.txt", parent_id="syn456").store()
+
+            File at the path `path/to/file.txt` and a parent folder with the ID `syn456`:
+
+                file_instance = File(path="path/to/file.txt").store(parent=Folder(id="syn456"))
+
+            Rename a file (Does not update the file on disk or the name of the downloaded file):
+
+                file_instance = File(id="syn123", download_file=False).get()
+                print(file_instance.name)  ## prints, e.g., "my_file.txt"
+                file_instance.change_metadata(name="my_new_name_file.txt")
+
+            Rename a file, and the name of the file as downloaded (Does not update the file on disk):
+
+                file_instance = File(id="syn123", download_file=False).get()
+                print(file_instance.name)  ## prints, e.g., "my_file.txt"
+                file_instance.change_metadata(name="my_new_name_file.txt", download_as="my_new_name_file.txt")
+
+        """
+        return self
+
+    def change_metadata(
+        self,
+        name: Optional[str] = None,
+        download_as: Optional[str] = None,
+        content_type: Optional[str] = None,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "File":
+        """
+        Change File Entity metadata for properties that are immutable after creation
+        through the store method.
+
+        Arguments:
+            name: Specify filename to change the filename of a file.
```

Comment:
nit: "...change the filename of a file as seen on Synapse."

Reply:
Great suggestion - Added: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1068/commits/b0227f9330b9b310b9e7c9bfc55fd6a9a181ca27